README.md

```diff
@@ -1,2 +1,238 @@
-# hclust
-Agglomerative hierarchical clustering in JavaScript
+### hclust
+[Agglomerative hierarchical clustering](https://en.wikipedia.org/wiki/Hierarchical_clustering) in JavaScript
+
+Adapted from [hcluster.js](https://github.com/cmpolis/hcluster.js) by [@ChrisPolis](https://twitter.com/chrispolis)
+
+---
+
+### Usage
+
+#### Browser
+
+```html
+<script src="hclust.min.js"></script>
+<script>
+  hclust.clusterData(...);
+  hclust.euclideanDistance(...);
+  hclust.avgDistance(...);
+  hclust.minDistance(...);
+  hclust.maxDistance(...);
+</script>
+```
+
+#### Node
+
+`npm install git+https://github.com/greenelab/hclust.git`
+
+or
+
+`yarn add git+https://github.com/greenelab/hclust.git`
+
+then 
+
+```javascript
+import { clusterData } from 'hclust';
+import { euclideanDistance } from 'hclust';
+import { avgDistance } from 'hclust';
+import { minDistance } from 'hclust';
+import { maxDistance } from 'hclust';
+```
+
+---
+
+### `clusterData({ data, key, distance, linkage, onProgress })`
+
+#### Parameters
+
+**`data`**
+
+The data you want to cluster, in the format:
+
+```javascript
+[
+  ...
+  [ ... 1, 2, 3 ...],
+  [ ... 1, 2, 3 ...],
+  [ ... 1, 2, 3 ...],
+  ...
+]
+```
+
+or if `key` parameter is specified:
+
+```javascript
+[
+  ...
+  { someKey: [ ... 1, 2, 3 ...] },
+  { someKey: [ ... 1, 2, 3 ...] },
+  { someKey: [ ... 1, 2, 3 ...] },
+  ...
+]
+```
+
+The entries in the outer array can be considered the `rows` and the entries within each `row` array can be considered the `cols`.
+Each `row` should have the same number of `cols`.
+
+*Default value:* `[]`
+
+**`key`**
+
+A `string` key to specify which values to extract from the `data` array.
+If omitted, `data` is assumed to be an array of arrays.
+If specified, `data` is assumed to be array of objects, each with a key that contains the values for that `row`.
+
+*Default value:* `''`
+
+**`distance`**
+
+A function to calculate the distance between two equal-dimension vectors, used in calculating the distance matrix, in the format:
+
+```javascript
+function (arrayA, arrayB) { return someNumber; }
+```
+
+The function receives two equal-length arrays of numbers.
```

Comment:
Just realized here - do you mean floating point or would integers be ok? Your examples above are floating point - just want to make sure there aren't any integer arithmetic gotchas in the implementation.

Reply:
The length of `arrayA` and `arrayB` must be the same, but `euclideanDistance` provided in the package can handle floats or ints in those arrays, and any function a user writes in place of `euclidean` should probably handle floats as well.

I've added a note to the readme to hopefully clarify this more.